Updates to ROTDIR/COMIN definitions related to compath.py and new GLDAS tag

KateFriedman-NOAA opened the pull request:
This PR includes changes to address feedback from NCO about how ROTDIR is set in the JJOB scripts, a correction to COMIN definitions, and two tag updates:

1. remove this block from all JJOB scripts:
```
if [ $RUN_ENVIR = "nco" ]; then
    export ROTDIR=${COMROOT:?}/${NET}/${gfs_ver}
fi
```
2. update `COMINwave` and `COMOUTwave` settings in `JGLOBAL_FORECAST` to use `compath.py` in the default setting, will use prior set `ROTDIR` from configuration level otherwise
3. change `ROTDIR` definition in `config.base.nco.static` (NCO mode) to use `compath.py`:
```
export ROTDIR="$(compath.py ${envir}/${NET}/${gfs_ver})"
```
4. update COMIN definitions to remove instances of `-o` after compath.py, add `${envir}`, and adjust line to compensate
5. update GLDAS tag to `gldas_gfsv16_release.v1.24.0`
6. update WAFS tag to `gfs_wafs.v6.2.7`
7. update GLDAS and WAFS tags in release notes

Including @WeiWei-NCO for feedback.

KateFriedman-NOAA committed:
Update ROTDIR in config.base.nco.static

- update ROTDIR setting in NCO mode base config to use compath.py in its
definition
- this change supports the removal of a RUN_ENVIR=nco if-block in the
JJOB scripts that set ROTDIR

Refs: #399

KateFriedman-NOAA committed:
Remove NCO if-block from JJOB scripts

- remove a RUN_ENVIR=nco if-block in JJOB scripts that set ROTDIR via
COMROOT
- ROTDIR is now set from the configuration level using compath.py
- also update COMIN[COMOUT]wave paths in JGLOBAL_FORECAST to use
compath.py for defaults

Refs: #399

KateFriedman-NOAA committed:
Update GLDAS tag to gldas_gfsv16_release.v1.24.0

Refs: #399

KateFriedman-NOAA committed:
Update GLDAS and WAFS tags in release notes

- update GLDAS tag to gldas_gfsv16_release.v1.24.0
- update WAFS to gfs_wafs.v6.2.6

Refs: #399

KateFriedman-NOAA committed:
Correct COMIN definitions

- remove "-o" after compath.py in COMIN definitions
- add "${envir}" and move closing ")" forward in line

Refs: #399

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2

KateFriedman-NOAA committed:
Update WAFS tag to gfs_wafs.v6.2.7

- update WAFS tag in sorc/checkout.sh and release notes

Refs: #399

KateFriedman-NOAA committed:
Merge branch 'feature/ops-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow into feature/ops-wcoss2

* 'feature/ops-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow:
  revert ecflow include files to NCO versions.  Will adapt as necessary for proper use

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2

WeiWei-NCO commented on jobs/JGDAS_ATMOS_GEMPAK_META_NCDC:
```diff
@@ -69,15 +69,15 @@ export DBNROOT=${DBNROOT:-${UTILROOT}/fakedbn}
 
 # export ukmet_ver=v2.2.0
-export COMINukmet=${COMINukmet:-$(compath.py -o ukmet/${ukmet_ver}/ukmet)}
+export COMINukmet=${COMINukmet:-$(compath.py ${envir}/ukmet/${ukmet_ver})}
```
I think it should be:
export COMINukmet=${COMINukmet:-$(compath.py ${envir}/ukmet/${ukmet_ver})/ukmet}
Because later in the scripts:
export HPCUKMET=${COMINukmet}.${PDY}

Same for COMINecmwf in the next line.

WeiWei-NCO replied:
compath.py call closed at ${ukmet_ver}, then append "/ukmet".

KateFriedman-NOAA replied:
Fixed, thanks!

WeiWei-NCO commented on jobs/JGFS_ATMOS_GEMPAK_META:
```diff
@@ -72,16 +72,16 @@ export DBN_ALERT_TYPE=GFS_METAFILE
 
 # export ukmet_ver=v2.2.0
-export COMINukmet=${COMINukmet:-$(compath.py -o ukmet/${ukmet_ver}/ukmet)}
+export COMINukmet=${COMINukmet:-$(compath.py ${envir}/ukmet/${ukmet_ver})}
```
ukmet, ecmwf, nam....
export COMINukmet=${COMINukmet:-$(compath.py ${envir}/ukmet/${ukmet_ver})/ukmet}
export COMINecmwf=${COMINecmwf:-$(compath.py ${envir}/ecmwf/${ecmwf_ver})/ecmwf}
export COMINnam=${COMINnam:-$(compath.py ${envir}/nam/${nam_ver})/nam}

KateFriedman-NOAA replied:
Fixed, thanks!

KateFriedman-NOAA committed:
Update COMIN paths for ukmet, ecmwf, and nam

- update COMIN paths in GEMPAK JJOB scripts for COMINukmet, COMINecmwf,
and COMINnam to add the respective systems to the end of the path
definition

Refs: #399

KateFriedman-NOAA committed:
Correct COMIN paths in GEMPAK driver scripts

Refs: #399

aerorahul commented on jobs/JGLOBAL_FORECAST:
```diff
@@ -111,13 +107,12 @@ status=$?
   export CDUMPwave=${CDUMPwave:-${CDUMP}wave}
-  export COMINwave=${COMINwave:-${ROTDIR}/${CDUMP}.${PDY}/${cyc}/wave}
-  export COMOUTwave=${COMOUTwave:-${ROTDIR}/${CDUMP}.${PDY}/${cyc}/wave}
+  export COMINwave=${COMINwave:-$(compath.py ${envir}/${NET}/${gfs_ver})/${CDUMP}.${PDY}/${cyc}/wave}
+  export COMOUTwave=${COMOUTwave:-$(compath.py -o ${NET}/${gfs_ver})/${CDUMP}.${PDY}/${cyc}/wave}
```
how is this going to affect development?

aerorahul reviewed (changes requested):
The extensive use of `compath.py` in the JJOB is going to cause a lot of `NCO` specific blocks.  JJOBS are agnostic to the machine or platform.  Using tools such as `compath.py` in a j-job is not permissible, since it does not exist on any other machine except WCOSS.

WeiWei-NCO commented:
One possible solution is to keep the compath.py in the J jobs but hardwire COM* in the config.base.emc.dyn. 

For this particular case,  COMINwave/COMOUTwave are defined in EMC config.base, so it won't affect dev runs.

In EMC config.base:
export COMINwave=${ROTDIR}/${CDUMP}.${PDY}/${cyc}/wave

Since config.base is called first, compath.py won't be called.
export COMINwave=${**COMINwave**:-$(compath.py ${envir}/${NET}/${gfs_ver})/${CDUMP}.${PDY}/${cyc}/wave}

WalterKolczynski-NOAA commented:
> The extensive use of `compath.py` in the JJOB is going to cause a lot of `NCO` specific blocks. JJOBS are agnostic to the machine or platform. Using tools such as `compath.py` in a j-job is not permissible, since it does not exist on any other machine except WCOSS.

IMO, as long as it is only there as a default if the variable is undefined, I think it's appropriate.
